src/driver/AmdCompiler.cpp

```diff
@@ -655,14 +676,61 @@ bool AMDGPUCompiler::LinkLLVMBitcode(const std::vector<Data*>& inputs, Data* out
     args.push_back(inputFile->Name().c_str());
   }
   File* outputFile = ToOutputFile(output, CompilerTempDir());
-  args.push_back("-o"); args.push_back(outputFile->Name().c_str());
-  for (const std::string& s : options) {
-    args.push_back(s.c_str());
+  if (IsInProcess() || IsLinkInProcess()) {
+    if (!options.empty()) {
+      std::vector<const char*> argv;
+      for (auto option : options) {
+        argv.push_back("");
+        argv.push_back(option.c_str());
+        // parse linker options
+        if (!cl::ParseCommandLineOptions(argv.size(), &argv[0], "llvm linker\n")) {
+          return false;
+        }
+        argv.clear();
+      }
+    }
+    LLVMContext context;
+    auto Composite = make_unique<llvm::Module>("composite", context);
+    Linker L(*Composite);
+    unsigned ApplicableFlags = Linker::Flags::None;
+    for (auto arg : args) {
+      SMDiagnostic error;
+      auto m = getLazyIRFileModule(arg, error, context);
+      if (!m.get()) {
+        errs() << "ERROR: the module '" << arg << "' loading failed!\n";
+        return false;
+      }
+      if (verifyModule(*m, &errs())) {
+        outs() << "ERROR: loaded module '" << arg << "' to link is broken!\n";
+        return false;
+      }
+      if (Verbose) {
+        std::cout << "Linking in '" << arg << "'\n";
+      }
+      if (L.linkInModule(std::move(m), ApplicableFlags)) {
```

Comment:
Did you solve the problem you had with huge linked files with unreferenced symbols?